### ISSUE TITLE
[CNN+LSTM] - Limit the number of layers in VGG encoder

### ISSUE THREAD

**Issue by JankowskiDaniel** [JankowskiDaniel/Neural-Deep-Retina#25]
Reduce the number of encoder layers to preserve the spatial information about the image (e.g. from (512, 1, 1) -> (32, 7, 7))

**Comment by JankowskiDaniel** [comment]
Here two options may be implemented:

- concatenate vectors
- aggregate vectors by e.g. mean

**Comment by JankowskiDaniel** [comment]
In the first experiment, I took only the first 10 layers (out of 31) from the VGG Encoder. The Encoder part architecture consists of 2 blocks:

Conv2D
ReLU
Conv2D
ReLU
MaxPool2D

The image spatial latent had the shape: [128, 12, 12].

The results on <b> test set </b> seems to be promising:

![test_outputs_and_targets](https://github.com/user-attachments/assets/d2b156d0-c7f8-4eed-bd24-4677cbb221ac)

However, a weird behavior was observed on testing the <b> train set </b>

![train_outputs_and_targets](https://github.com/user-attachments/assets/94f2f1a5-0fa0-4a31-9f22-a6ae42f1153b)

It's weird as the loss function decreased significantly and regularly during the training.
